tests/parsers/test_orca.py

```diff
@@ -26,3 +26,49 @@ def test_orca_default(aiida_localhost, generate_calc_job_node, generate_parser,
         'relaxed_structure': structure_attributes,
         'output_parameters': results['output_parameters'].attributes,
     })
+
+
+def test_orca_unsuccessfull(
+    aiida_localhost, generate_calc_job_node, generate_parser, generate_inputs_orca, data_regression
+):
+    """Test that the parser returns non-zero exit code in case of unsuccessfull ORCA job."""
+    name = 'unsuccessfull'
+    entry_point_calc_job = 'orca.orca'
+    entry_point_parser = 'orca_base_parser'
+
+    node = generate_calc_job_node(entry_point_calc_job, aiida_localhost, name, generate_inputs_orca())
+    parser = generate_parser(entry_point_parser)
+    results, calcfunction = parser.parse_from_node(node)
+
+    assert calcfunction.is_finished, calcfunction.exception
+    assert calcfunction.exit_status > 299
+    assert not calcfunction.is_finished_ok, calcfunction.exit_message
+    assert 'relaxed_structure' in results
+    assert 'output_parameters' in results
+
+    structure_attributes = results['relaxed_structure'].attributes
+
+    # Pop the cell if it is there, ORCA does not support periodic cell calculations
+    structure_attributes.pop('cell', None)
+
+    data_regression.check({
+        'relaxed_structure': structure_attributes,
+        'output_parameters': results['output_parameters'].attributes,
+    })
+
+
+def test_orca_missing_stdout(aiida_localhost, generate_calc_job_node, generate_parser, generate_inputs_orca):
```

Comment:
This new test is not strictly related to this PR, but I though it is useful to add to have more coverage of the parser error paths. It checks the following code

https://github.com/pzarabadip/aiida-orca/blob/4a3737a98f8458837ebfaf81d1dd9eb1c41ea953/aiida_orca/parsers/__init__.py#L40